### ISSUE TITLE
godot 4 pico 4 movement provider not working

### ISSUE THREAD

**Issue by flobotics** [GodotVR/godot-xr-tools#297]
hi,
could it be that the direct movement provider for pico 4 does not work , or do i miss something ?

thx

**Comment by Malcolmnixon** [comment]
The Direct Movement provider just takes the inputs from the XR system mapped through the OpenXR Action Map. I'm not seeing a Pico 4 profile in the OpenXR Action Map, but by default the Direct Movement provider reads its input from the Vector2 input called "primary".

If you can track down what the action is called for the Pico, you could try changing the "Input Action" property of the Direct Movement provider to match.

**Comment by BastiaanOlij** [comment]
Yeah PICO controller mappings can't be added yet and PICO will fall back to the `khr/simple_controller` which doesn't have joystick inputs.

We're waiting on the PICO interaction extension to become an official part of OpenXR so we can include bindings for it.

**Comment by BastiaanOlij** [comment]
Ok, so this won't be merged until we have updated OpenXR headers that include the PICO vendor extension but in the meanwhile you can build Godot with this PR included for proper PICO support: https://github.com/godotengine/godot/pull/70167

**Comment by flobotics** [comment]
hi @BastiaanOlij ,

does this PR will get included in the next godot-4 beta release ?

**Comment by BastiaanOlij** [comment]
@flobotics not likely, as I said we're waiting on the official OpenXR spec being updated to include the PICO controller extension, this could take awhile.

**Comment by BastiaanOlij** [comment]
@flobotics , woot woot, we managed to solve this differently with some long awaited enhancements to the way we handle extensions so the PR got merged today. It will be in the next beta.

Do note that currently PICO only works with OpenGL, we're still waiting on an OS patch on the PICO before it will work with Vulkan and that will likely coincide with PICO being added to the official OpenXR specs.
